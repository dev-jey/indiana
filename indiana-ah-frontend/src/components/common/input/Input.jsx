import styled from 'styled-components';
// Create an Input component that'll render an <input> tag with some styles
const Input = styled.input`
  display: flex;
  padding: 0.8rem;
  margin: ${props => props.margin || '0.1rem auto'};
  height: 4.3rem;
  fill: #fbfcff;
  border: 0.05rem solid rgba(0, 0, 0, 0.4);
  border-radius: 0.4rem;
  width: ${props => props.inputWidth || '94%'};
  box-sizing: border-box;
  padding: 0.8rem 1.9rem;
  height: 4.3rem;
  background: #ffffff;
  border: 0.05rem solid rgba(0, 0, 0, 0.1);
  border-radius: 0.2rem;
  @media (max-width: 76.8rem) {
    width: 100%;
    padding: 0.8rem;
    flex-direction: column;
    height: 3.5rem;
    font-size: 1.5rem;
  }
`;

const TextArea = styled.textarea`
  padding: 0.8rem 1.9rem;
  background: #ffffff;
  border: 0.05rem solid rgba(0, 0, 0, 0.1);
  border-radius: 0.2rem;
  width: ${props => props.inputWidth || '37.1rem'};
  box-sizing: border-box;
`;

export { Input, TextArea };
