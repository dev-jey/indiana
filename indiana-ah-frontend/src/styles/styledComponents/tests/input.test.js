import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import InputField from '../../../components/common/input/InputComponent.jsx';
import { Input, TextArea } from '../../../components/common/input/Input.jsx';

describe('Custom Input component', () => {
  const props = {
    type: 'email',
    id: 'email',
    value: 'Odinks',
    placeholder: 'Enter your email',
    handleChange: () => 2
  };
  const wrapper = shallow(<InputField {...props} />);
  it('should create an instance and render correctly', () => {
    expect(wrapper.length).toEqual(1);
    expect(wrapper.find('div')).toHaveLength(1);
    wrapper.setProps({ ...props });
  });
});

describe('Custom Input styled-component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Input />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders dynamic width', () => {
    const tree = renderer.create(<Input inputWidth="20rem" />).toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule('width', expect.stringContaining('20rem'));
  });
});
describe('Custom Text Area styled-component', () => {
  it('renders dynamic width', () => {
    const tree = renderer.create(<TextArea inputWidth="20rem" />).toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule('width', expect.stringContaining('20rem'));
  });
  it('renders dynamic width', () => {
    const tree = renderer.create(<TextArea />).toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule('width', expect.stringContaining('37.1rem'));
  });
});
