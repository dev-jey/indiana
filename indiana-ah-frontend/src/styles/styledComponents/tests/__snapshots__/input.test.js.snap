// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom Input styled-component renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0.8rem;
  margin: 0.1rem auto;
  height: 4.3rem;
  fill: #fbfcff;
  border: 0.05rem solid rgba(0,0,0,0.4);
  border-radius: 0.4rem;
  width: 94%;
  box-sizing: border-box;
  padding: 0.8rem 1.9rem;
  height: 4.3rem;
  background: #ffffff;
  border: 0.05rem solid rgba(0,0,0,0.1);
  border-radius: 0.2rem;
}

@media (max-width:76.8rem) {
  .c0 {
    width: 100%;
    padding: 0.8rem;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    height: 3.5rem;
    font-size: 1.5rem;
  }
}

<input
  className="c0"
/>
`;

exports[`Custom Input styled-component renders dynamic width 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0.8rem;
  margin: 0.1rem auto;
  height: 4.3rem;
  fill: #fbfcff;
  border: 0.05rem solid rgba(0,0,0,0.4);
  border-radius: 0.4rem;
  width: 20rem;
  box-sizing: border-box;
  padding: 0.8rem 1.9rem;
  height: 4.3rem;
  background: #ffffff;
  border: 0.05rem solid rgba(0,0,0,0.1);
  border-radius: 0.2rem;
}

@media (max-width:76.8rem) {
  .c0 {
    width: 100%;
    padding: 0.8rem;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    height: 3.5rem;
    font-size: 1.5rem;
  }
}

<input
  className="c0"
/>
`;

exports[`Custom Text Area styled-component renders dynamic width 1`] = `
.c0 {
  padding: 0.8rem 1.9rem;
  background: #ffffff;
  border: 0.05rem solid rgba(0,0,0,0.1);
  border-radius: 0.2rem;
  width: 20rem;
  box-sizing: border-box;
}

<textarea
  className="c0"
/>
`;

exports[`Custom Text Area styled-component renders dynamic width 2`] = `
.c0 {
  padding: 0.8rem 1.9rem;
  background: #ffffff;
  border: 0.05rem solid rgba(0,0,0,0.1);
  border-radius: 0.2rem;
  width: 37.1rem;
  box-sizing: border-box;
}

<textarea
  className="c0"
/>
`;
